com.chartboost.helium/Android~/helium-android-bridge/src/main/kotlin/com/chartboost/heliumsdk/unity/HeliumEventProcessor.kt

```diff
@@ -0,0 +1,95 @@
+package com.chartboost.heliumsdk.unity
+
+import android.util.Log
+import com.chartboost.heliumsdk.ad.HeliumAdError
+import org.json.JSONException
+import org.json.JSONObject
+
+object HeliumEventProcessor {
+    private const val TAG = "HeliumEventProcessor"
+    private const val EMPTY_STRING = ""
+
+    @JvmStatic
+    fun serializeHeliumEvent(
+        placement: String?,
+        heliumAdError: HeliumAdError?,
+        eventConsumer: HeliumEventConsumer<String?, Int, String?>
+    ) {
+        var placementName: String? = EMPTY_STRING
+        if (placement != null) placementName = placement
+        var errorCode = -1
+        var errorDescription = EMPTY_STRING
+        if (heliumAdError != null) {
+            val tempCode = heliumAdError.getCode()
+            errorDescription = heliumAdError.getMessage()
+
+            // we do this in order to bypass lint issues
+            if (errorCode != tempCode) errorCode = tempCode
+        }
+        eventConsumer.accept(placementName, errorCode, errorDescription)
+    }
+
+    @JvmStatic
+    fun serializeHeliumBidEvent(
+        placement: String?,
+        dataMap: HashMap<String, String>,
```

Comment:
it is recommended to only specify `Map` and not specifically a `HashMap` as there's no reason to.

Reply:
This is because the value being specified by the callback is a HashMap itself, I am sure we could do the conversion but also this is a very specific setting use case so I don't see the need to change it either.